symengine/symengine_casts.h

```diff
@@ -33,7 +33,7 @@ struct remove_reference<T &> {
 template <typename To, typename From>
 inline To implicit_cast(const From &f)
 {
-    return f;
+    return static_cast<To>(f);
```

Comment:
I don't think that's the correct fix. I think this should be left as it was, to allow the compiler to check types.